File: src/p1108/golang/solution.go
func defangIPaddr(address string) string {
	var result []byte
	for i := 0; i < len(address); i++ {
		if address[i] == '.' {
			result = append(result, '[', '.', ']')
		} else {
			result = append(result, address[i])
		}
	}
	return string(result)
}

File: src/p1108/cpp/solution.cpp
class Solution {
public:
    string defangIPaddr(const string &address) {
        string result;
        for (auto x : address) {
            if (x == '.') {
                result.append("[.]");
            } else {
                result.push_back(x);
            }
        }
        return result;
    }
};
